pkg/resources/task.go

```diff
@@ -2,709 +2,648 @@ package resources
+	"error_integration": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		DiffSuppressFunc: SuppressIfAny(suppressIdentifierQuoting, IgnoreChangeToCurrentSnowflakeValueInShow("error_integration")),
+		Description:      blocklistedCharactersFieldDescription("Specifies the name of the notification integration used for error notifications."),
 	},
 	"comment": {
 		Type:        schema.TypeString,
 		Optional:    true,
 		Description: "Specifies a comment for the task.",
 	},
+	"finalize": {
+		Optional:         true,
+		Type:             schema.TypeString,
+		ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		DiffSuppressFunc: SuppressIfAny(
+			suppressIdentifierQuoting,
+			IgnoreChangeToCurrentSnowflakeValueInShow("task_relations.0.finalized_root_task"),
+		),
+		Description:   blocklistedCharactersFieldDescription("Specifies the name of a root task that the finalizer task is associated with. Finalizer tasks run after all other tasks in the task graph run to completion. You can define the SQL of a finalizer task to handle notifications and the release and cleanup of resources that a task graph uses. For more information, see [Release and cleanup of task graphs](https://docs.snowflake.com/en/user-guide/tasks-graphs.html#label-finalizer-task)."),
+		ConflictsWith: []string{"schedule", "after"},
+	},
 	"after": {
-		Type:          schema.TypeList,
-		Elem:          &schema.Schema{Type: schema.TypeString},
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			DiffSuppressFunc: suppressIdentifierQuoting,
+			ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		},
 		Optional:      true,
-		Description:   "Specifies one or more predecessor tasks for the current task. Use this option to create a DAG of tasks or add this task to an existing DAG. A DAG is a series of tasks that starts with a scheduled root task and is linked together by dependencies.",
-		ConflictsWith: []string{"schedule"},
+		Description:   blocklistedCharactersFieldDescription("Specifies one or more predecessor tasks for the current task. Use this option to [create a DAG](https://docs.snowflake.com/en/user-guide/tasks-graphs.html#label-task-dag) of tasks or add this task to an existing DAG. A DAG is a series of tasks that starts with a scheduled root task and is linked together by dependencies."),
+		ConflictsWith: []string{"schedule", "finalize"},
 	},
 	"when": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Description: "Specifies a Boolean SQL expression; multiple conditions joined with AND/OR are supported.",
+		Type:             schema.TypeString,
+		Optional:         true,
+		DiffSuppressFunc: SuppressIfAny(DiffSuppressStatement, IgnoreChangeToCurrentSnowflakeValueInShow("condition")),
+		Description:      "Specifies a Boolean SQL expression; multiple conditions joined with AND/OR are supported.",
```

Comment:
I think this description is incomplete: It's not clear what `when` it's used for.